### ISSUE TITLE
Cachetool recipe no more compatible with php <7.1

### ISSUE THREAD

**Issue by hyde1** [deployphp/recipes#183]
| Q                 | A
| ----------------- | ---
| Issue Type        | Bug
| Deployer Version  | 6.2.0
| Recipes Version   | 6.1.1
| Local Machine OS  | Ubuntu 16.04.4 LTS with php7.0
| Remote Machine OS | Ubuntu 16.04.4 LTS with php7.0

### Description 
Recipe cachetool is no more compatible with php 7.0. Needs php 7.1.
See https://github.com/gordalina/cachetool/issues/73 :

> Current version, 4.0.0 supports only php 7.1 and symfony 4
> Version 3.2.1 supports php>=5.5, you can download it here: 

### Steps to reproduce
1. Use 

### Content of `deploy.php`
```php
...
require 'recipe/cachetool.php';
after('deploy:symlink', 'cachetool:clear:opcache');
...
```

### Output log
```
In Client.php line 96:
 
  The command "cd /var/phpapp/releases/393 && (/usr/bin/php cachetool.phar opcache:reset --fcgi=/run/php/php7.0-fpm.sock)" failed. 

  Exit Code: 255 (Unknown error)
    
  Host Name: hostname
  ================                                                                                                                                  
  PHP Parse error:  syntax error, unexpected '?' in phar:///var/phpapp/releases/393/cachetool.phar/vendor/symfony/yaml/Parser.php on line 499 

```

**Comment by antonmedv** [comment]
@pschirch update cachetool.phar

**Comment by pschirch** [comment]
@antonmedv We need a specific cachetool.phar version (3.2.1). Latest only works on PHP >= 7.1. Unfortunately the download source is [hard coded](https://github.com/deployphp/recipes/blob/c15a8e8a2d75e0ba61ec03fcf99c92bf6c0ef2ce/recipe/cachetool.php#L47).

**Comment by antonmedv** [comment]
You can put a proper cachetool.phar version by hands. Recipe checks if there is a cachetool.phar before downloading newer.

**Comment by jamieburchell** [comment]
Is this the same problem - I've started to see this within the last few hours:

` [Deployer\Exception\RuntimeException]
  The command "cd /var/www/vhosts/aslotel.co.uk/releases/17 && (/usr/bin/php cachetool.phar opcache:reset --fcgi=127.0.0.1:9000)" failed.
  Exit Code: 255 (Unknown error)
  Host Name: 46.101.80.231
  ================
  PHP Fatal error:  Default value for parameters with a class type hint can only be NULL in phar:///var/www/vhosts/x.co.uk/releases/17/cachetool.phar/vendor/symfony/yaml/Parser.php on line 49`

Basically all our deployments are failing.

**Comment by pschirch** [comment]
@hyde1 @antonmedv 

Any updates?

@jamieburchell 

You can temporarily define a task for installing a specific CacheTool version.

```php
# *****************************************************************************
# * CacheTool Installer
# *****************************************************************************
desc('CacheTool Installer');
task('cachetool:install', function () {
    $releasePath = get('release_path');
    cd($releasePath);
    run("curl -s https://gordalina.github.io/cachetool/downloads/cachetool-3.2.1.phar --output cachetool.phar");
});
```

**Comment by hyde1** [comment]
Hi @antonmedv what do you think of my PR?

**Comment by jamieburchell** [comment]
@pschirch Thanks. I spotted that we can download the phar and it'll use that, so I've been doing that just to get things working again.

**Comment by antonmedv** [comment]
Hi, sorry for delay. Will merge it today.

**Comment by antonmedv** [comment]
Done and released.

**Comment by jamieburchell** [comment]
Thank you :)

**Comment by llaumgui** [comment]
Not fixed for me on PHP 7.2:
~~~
  [Deployer\Exception\RuntimeException]                                                                                                                     
  The command "cd /var/www/xxx.yyy.fr/releases/43 && (/usr/local/bin/php -r "echo version_compare(phpversion(), '7.1') == -1 ? 'cachetool-3.2.1.phar' : 'cachetool.phar';")" failed.                                                                                                                        
                                                                                                                                                            
  Exit Code: 2 (Misuse of shell builtins)                                                                                                                   
                                                                                                                                                            
  Host Name: xxx.yyy.fr                                                                                                                             
                                                                                                                                                            
  ================                                                                                                                                          
  /bin/sh: syntax error: unexpected "("   
~~~

**Comment by antonmedv** [comment]
Try run failed command on server via ssh.

**Comment by llaumgui** [comment]
Hum, problem is my php binary whitch map my docker container. Sorry.
